agent-lib/src/main/java/org/terracotta/angela/agent/Agent.java

```diff
@@ -63,6 +68,12 @@ public class Agent implements AutoCloseable {
   private static final Logger logger;
 
   static {
+    // Required to avoid a deadlock if a client job causes a system.exit to be run.
+    // The agent has its own shutdown hook
+    System.setProperty(IgniteSystemProperties.IGNITE_NO_SHUTDOWN_HOOK, "true");
+
+    System.setProperty(IgniteSystemProperties.IGNITE_UPDATE_NOTIFIER, "false");
```

Comment:
what is the use of this ?

Reply:
oh got this its pop up to notify to update ignite

Reply:
IGNITE_UPDATE_NOTIFIER => do not check on internet for a new Ignite version each time.
IGNITE_NO_SHUTDOWN_HOOK => do not tell ignite to register a shutdown hook to close ignite because we are already doing that in our shutdown hook.

Note: This only applies to launched Agent.

Reply:
Note: I have moved them in the ignite() method, the IGNITE_UPDATE_NOTIFIER was already there ;-)